src/domain/safe-apps/entities/safe-app-access-control.entity.ts

```diff
@@ -1,3 +1,10 @@
+export enum SafeAppAccessControlPolicies {
+  NoRestrictions = 'NO_RESTRICTIONS',
+  DomainAllowlist = 'DOMAIN_ALLOWLIST',
+  Unknown = 'UNKNOWN',
+}
+
 export interface SafeAppAccessControl {
   type: string;
```

Comment:
For the `type` we can use the enum itself:

```typescript
type: SafeAppAccessControlPolicies;
```

Reply:
> Hmm, I had that type before but later I changed it because I thought this is coming from an external service and therefore we shouldn't assume it would match a `SafeAppAccessControlPolicy`.

The responsibility of the parsing and validation is up to AJV and this being a domain entity could be more strict in the values that it holds. However an argument can be made that we cannot distinguish between types easily without the parameter `type`.

> In fact, there's a use case where a random string is received, and it gets mapped to `UNKNOWN`. If we map this to the enum directly, we couldn't test it (TS wouldn't accept a random string in [this test](https://github.com/safe-global/safe-client-gateway-nest/blob/67da9f6389261d3e48bcc6ff94436a983bfe1d9d/src/routes/safe-apps/safe-apps.controller.spec.ts#L105), by instance).

I don't think this is correct. You can't test it if you use the enum but you can return any json payload in the tests (i.e. without a builder).

---

No need to tackle it here. It might not be straightforward to distinguish between the different types without that parameter. Type predicates won't help much either here I think.

Reply:
> I don't think this is correct. You can't test it if you use the enum but you can return any json payload in the tests (i.e. without a builder).

Yes, you're right, I didn't realize I could modify the response directly instead of using the builder so we can type the domain entity 👍🏻  I've modified it in [e3a8849](https://github.com/safe-global/safe-client-gateway-nest/pull/383/commits/e3a88490d0d5cb5623bb06a2c337f7d87c028615), thanks!